var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -24,7 +25,7 @@ class Mvapich2(AutotoolsPackage):
     version('2.1', sha256='49f3225ad17d2f3b6b127236a0abdc979ca8a3efb8d47ab4b6cd4f5252d05d29')
 
     provides('mpi')
-    provides('mpi@:3.0')
+    provides('mpi@:3.1')
```

Comment:
Do prior versions provide lower levels?  If so this should probably be updated to, e.g.:

```python
provides('mpi@:3.0' when='@2.1:')
provides('mpi@:3.1' when='@2.2:')
```

etc.  Look at `mpich` for examples.

Reply:
Hi, Todd.

We have been at MPI 3.1 for a couple of years now. I will add older versions.

Thx,
hari.

Reply:
```
    Update different provides for older versions based on feedback from Todd Gamblin

diff --git a/var/spack/repos/builtin/packages/mvapich2/package.py b/var/spack/repos/builtin/packages/mvapich2/package.py
index f172522..326cdea 100644
--- a/var/spack/repos/builtin/packages/mvapich2/package.py
+++ b/var/spack/repos/builtin/packages/mvapich2/package.py
@@ -25,7 +25,9 @@ class Mvapich2(AutotoolsPackage):
     version('2.1', sha256='49f3225ad17d2f3b6b127236a0abdc979ca8a3efb8d47ab4b6cd4f5252d05d29')

     provides('mpi')
-    provides('mpi@:3.1')
+    provides('mpi@:3.1', when='@2.3:')
+    provides('mpi@:3.0', when='@2.2')
+    provides('mpi@:3.0', when='@2.1')

     variant('debug', default=False,
             description='Enable debug info and error messages at run-time')
```

Does this look OK?

Reply:
See other comment -- you can do this with only one rule for mpi 3.